src/rgw/rgw_op.cc

```diff
@@ -603,7 +603,7 @@ rgw::IAM::Environment rgw_build_iam_environment(RGWRados* store,
     e.emplace("aws:SecureTransport", "true");
   }
 
-  i = m.find("HTTP_HOST");
+  i = m.find("REMOTE_ADDR");
```

Comment:
This apparently doesn't work on all backends. I don't know if we're eventually moving to All Beast All the Time, but for now make sure to adapt to whatever backend we're using.